core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -47,6 +40,12 @@
  */
 public class RoutingNodes implements Iterable<RoutingNode> {
 
+    private final static Random RANDOM;
+
+    static {
+        RANDOM = Randomness.create();
+    }
+
```

Comment:
Just to make it better, setting it here in clinit is not so ideal, because its not perfectly reproducible when multiple tests are run in the same jvm. you will still have perceived reproducibility issues vs jenkins if you go about it this way: because the random will be initialized *once* and then we will run 8 test classes against it.

then, when you later try to reproduce the one that failed, the sequence will be different (even though the initial value is the same), because the other tests are not also invoked.

but if you do it this way, it at least allows "whole build" reproducibility, which is an improvement. That means e.g. if you nuke your local execution hints file and run 'gradle test -Dtests.seed=xxxxxx -Dtests.jvm=yyyyyy', it will match what jenkins did. But nobody does that.